src/basic/log.c

```diff
@@ -49,6 +49,7 @@ static int kmsg_fd = -1;
 static int journal_fd = -1;
 
 static bool syslog_is_stream = false;
+bool found_error_when_parsing_config = false; /* to detect when unit verification fails */
```

Comment:
I think the comment should be in the .h file. It should also explain why this is a global variable.

Right now the name of the variable describes how you are using it, but not what actual meaning of the variable is. It is set when `log_syntax` is used. `log_syntax` can be called from various other places, not just config parsing.  A name like `log_syntax_issue_reported` would be better. (It also has the advantage that it is "namespaced" properly, i.e. the reader can easily guess that it's declared in `log.h`.)